packages/financial-templates-lib/src/price-feed/HmLandRegistryPriceFeed.ts

```diff
@@ -0,0 +1,253 @@
+
+    // 1. Construct URL & SPARQL query.
+    
+    const url = "http://landregistry.data.gov.uk/landregistry/query";
+    
+    const sparqlQuery = `
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+PREFIX sr: <http://data.ordnancesurvey.co.uk/ontology/spatialrelations/>
+PREFIX ukhpi: <http://landregistry.data.gov.uk/def/ukhpi/>
+PREFIX lrppi: <http://landregistry.data.gov.uk/def/ppi/>
+PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+PREFIX lrcommon: <http://landregistry.data.gov.uk/def/common/>
+
+# House price index for a specific region within a given date range
+SELECT ?region ?date ?hpi
+{
+  ?obs ukhpi:refRegion ?region ;
+       ukhpi:refPeriodStart ?date ;
+       ukhpi:housePriceIndex ?hpi .
+
+  FILTER (
+    ?date > "${startDateTimeString}"^^xsd:date &&
+    ?date <= "${endDateTimeString}"^^xsd:date
+  )
+
+  FILTER (?region = <http://landregistry.data.gov.uk/id/region/united-kingdom>)
+}
+ORDER BY DESC(?date)
+`;
+
+    const params = `query=${encodeURIComponent(sparqlQuery)}&output=json`;
+
+    // 2. Send request.
+    const historyResponse = await this.networker.getJson(`${url}?${params}`);
+    console.log("historyResponse", historyResponse);
+    // Sample Response
+    // {
+    //    "head": {
+    //      "vars": [ "region" , "date" , "hpi" ]
+    //     } ,
+    //      "results": {
+    //        "bindings": [
+    //          {
+    //            "region": { "type": "uri" , "value": "http://landregistry.data.gov.uk/id/region/united-kingdom" } ,
+    //            "date": { "type": "literal" , "datatype": "http://www.w3.org/2001/XMLSchema#date" , "value": "2023-02-01" } ,
+    //            "hpi": { "type": "literal" , "datatype": "http://www.w3.org/2001/XMLSchema#decimal" , "value": "150.79" }
+    //          }
+    //        ]
+    //      }
+    //    }
+
+
+    // 3. Check responses.
+    if (!historyResponse?.results.bindings || historyResponse.results.bindings.length === 0) {
+      throw new Error(`🚨Could not parse price result from url ${url}: ${JSON.stringify(historyResponse)}`);
+    }
+
+    // 4. Parse results.
+    // historyResponse.results
+    const newHistoricalPricePeriods =
+      historyResponse.results.bindings
+        .map((dailyData: any) => {
+          return {
+            date: this._dateTimeToSecond(dailyData.date.value),
+            value: this.convertPriceFeedDecimals(dailyData.hpi.value),
+          }
+        });
+
+     // checks if the UKHPI value is being returned correctly   
+      newHistoricalPricePeriods.forEach((item: any) => {
+          console.log('hpi value:', item.value.toString());
+      });
+
+
+    // 5. Store results.
+    this.currentPrice = newHistoricalPricePeriods[newHistoricalPricePeriods.length - 1].value;
+    this.priceHistory = newHistoricalPricePeriods;
+    this.lastUpdateTime = currentTime;
```

Comment:
I tested the price feed be adding these lines here, and giving different lookbacks like 120 days, 365 days etc.

```
    // TESTING Update price function
    console.log("DEBUG getCurrentPrice: ", this.getCurrentPrice()?.toString());
    console.log("DEBUG getLastUpdateTime: ", this.getLastUpdateTime());
    console.log("DEBUG getHistoricalPrice: 1st May", (await this.getHistoricalPrice(1682928606))?.toString());
    console.log("DEBUG getHistoricalPrice: 5th March", (await this.getHistoricalPrice(1678003806))?.toString());
    console.log("DEBUG getHistoricalPrice: 9th Feb", (await this.getHistoricalPrice(1675930206))?.toString());
    console.log("DEBUG getHistoricalPrice: 1st Jan", (await this.getHistoricalPrice(1672560606))?.toString());
    console.log("DEBUG getHistoricalPrice: 28th Dec", (await this.getHistoricalPrice(1672215006))?.toString());
```

Reply:
I repeated this exact testing process locally after making the above changes you recommended and everything appears to be working perfectly now. I think it is good to merge :)

Reply:
Perfect 💯